### DISCUSSION TITLE
error:delete a data point

### DISCUSSION THREAD

**Discussion by Chrisable** [pyg-team/pytorch_geometric#4812] [General]
Firstly,i want to delete a data point in this way:
```
x1 = x[0:1358]
x2 = x[1359:data.num_nodes]
new_x = torch.cat((x1,x2), dim=0)

y1 = y[0:1358]
y2 = y[1359:data.num_nodes]
new_y = torch.cat((y1,y2), dim=0)


row_index1 = (edge_index[0] == 1358)
col_index1 = (edge_index[1] == 1358)
index = torch.logical_not(torch.logical_or(row_index1,col_index1))
new_edge_index = edge_index[:,index]

train_mask1 = data.train_mask[0:1358]
train_mask2 = data.train_mask[1359:data.num_nodes]
new_train_mask = torch.cat((train_mask1, train_mask2), dim=0)

val_mask1 = data.val_mask[0:1358]
val_mask2 = data.val_mask[1359:data.num_nodes]
new_val_mask = torch.cat((val_mask1, val_mask2), dim=0)

test_mask1 = data.test_mask[0:1358]
test_mask2 = data.test_mask[1359:data.num_nodes]
new_test_mask = torch.cat((test_mask1, test_mask2), dim=0)
data1 = Data(x=new_x,edge_index=new_edge_index,y=new_y,train_mask=new_train_mask,val_mask=new_val_mask,test_mask=new_test_mask)
```

when i train a gat model on the data1,it exists a error:
![image](https://user-images.githubusercontent.com/49330390/174088637-487a1a2e-2ec8-4b75-a038-4bc4b0277050.png)

**Comment by rusty1s** [comment]
You should use `data.subgraph` for this and pass a mask to it with element 1358 being discarded. The issue in your code is that you do not decrease the values in `edge_index`.
